packages/gatsby-source-contentful/src/source-nodes.js

```diff
@@ -57,61 +57,39 @@ export async function sourceNodes(
   },
   pluginOptions
 ) {
-  const hasStatefulSourceNodes = hasFeature(`stateful-source-nodes`)
-  const needToTouchNodes = !hasStatefulSourceNodes
-
   const {
-    createNode,
+    createNode: originalCreateNode,
     touchNode,
-    deleteNode,
+    deleteNode: originalDeleteNode,
     unstable_createNodeManifest,
-    enableStatefulSourceNodes,
   } = actions
 
-  const online = await isOnline()
+  const pluginConfig = createPluginConfig(pluginOptions)
 
-  if (hasStatefulSourceNodes) {
-    enableStatefulSourceNodes()
-  }
-  // Gatsby only checks if a node has been touched on the first sourcing.
-  // As iterating and touching nodes can grow quite expensive on larger sites with
-  // 1000s of nodes, we'll skip doing this on subsequent sources.
```

Comment:
Why is this removed?

Reply:
It's inside `getExistingCachedNodes` now so we can iterate through all nodes 1 fewer time

Reply:
I also added a new API `enableStatefulSourceNodes` which makes it so touching nodes isn't needed when that feature is supported. It's nice cause without that Gatsby loads every node into memory once again when touching nodes 😅so this way on newer Gatsby versions a lot less memory is used